### PR TITLE
Andrew Dowell (Ronin)

### PR DESCRIPTION
**Pull request by k0zakinio**

### CODE REVIEW

File: app/helpers/comments_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module CommentsHelper
```
Comment (yevdyko) [inline app/helpers/comments_helper.rb]: Should be deleted if not needed.

File: app/helpers/likes_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module LikesHelper
```
Comment (yevdyko) [inline app/helpers/likes_helper.rb]: Should be deleted if not needed.

File: app/helpers/uploads_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module UploadsHelper
```
Comment (yevdyko) [inline app/helpers/uploads_helper.rb]: Should be deleted if not needed.

File: spec/features/comments_feature_spec.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+
+feature 'Comments' do
+  context 'Adding a comment' do
+    scenario 'can add a comment' do
+      visit '/'
+      expect(page).not_to have_content("Add comment")
```
Comment (yevdyko) [inline spec/features/comments_feature_spec.rb]: It's better to separate out tests

File: spec/features/like_feature_spec.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+feature 'Likes' do
+  before do
+    sign_up('adowellphoto@gmail.com', 'password')
+    upload_image('One of my favorites', true)
+  end
+  context 'Singular likes' do
+    scenario 'I can like an upload' do
+      visit '/'
+      expect(page).to have_content('0 Likes')
```
Comment (yevdyko) [inline spec/features/like_feature_spec.rb]: It's better to separate out tests

File: app/controllers/comments_controller.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+class CommentsController < ApplicationController
+
+  before_action :authenticate_user!
+
+  def create
+    @upload = Upload.find(params[:upload_id])
```
Comment (yevdyko) [inline app/controllers/comments_controller.rb]: If you are redirecting, you do not need to set @ variables

File: app/controllers/likes_controller.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+class LikesController < ApplicationController
+
+  before_action :authenticate_user!
+
+  def create
```
Comment (yevdyko) [inline app/controllers/likes_controller.rb]: If you are redirecting, you do not need to set @ variables.

File: app/controllers/uploads_controller.rb [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+  def index
+    @uploads = Upload.order(id: :DESC).all
+  end
+
+  def new
+    @upload = Upload.new
+  end
+
+  def create
+    @upload = Upload.new(upload_params)
```
Comment (yevdyko) [inline app/controllers/uploads_controller.rb]: If you are redirecting, you do not need to set @ variables.

File: app/views/uploads/new.html.erb [hunk 9]
```diff
@@ -0,0 +1,6 @@
+<%= form_for @upload, html: {multipart: true} do |f| %>
```
Comment (yevdyko) [inline app/views/uploads/new.html.erb]: Can be replaced with local variable.

**Comment by tansaku** [comment]
You had your instagram_challenge reviewed by **Yev Dyko**.
### The good points are:
* Correct installation instructions in README, Correct usage instructions in README
* Consistent indentation
* Text expectations appropriate, Test descriptions consistent, Tests DRY
* Models appropriately tested
* Feature tests declarative
* Nested forms used appropriately
* All controller methods short and succinct
* Thin controllers with minimal logic
* Model logic is DRY
* Models appropriately connected
* Relative and absolute paths used appropriately
* Asset Path used correctly, Responsive images supported with CSS
* Long nested call chains avoided

### You should consider the following improvements:
* Ensure Rakefile has Appropriate Tasks
* Gemfile should use Test Groups
* Ensure spec_helper.rb is Configured Correctly
* Ensure Asset Routes are Set Correctly
* Delete empty helper files
* No need to set instance vars when redirecting
* Feature tests should use 'within' for precision
* Tests should separate out expectations where appropriate
* Use partials appropriately
* Replace instance variables with local variables
* Prefer other Semantic HTML Elements to Divs Where Possible
* Avoid anonymous javascript functions

### Additional comments:
Really good job!




see https://github.com/makersacademy/instagram_challenge/blob/master/docs/review.md for more details on the items above

**Pull request closed without merging**